packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -5558,30 +5506,30 @@ async function setFakeProvider(
  * @returns A promise that resolves to the list of payloads for the set of
  * events, optionally filtered, when a specific number of them have occurred.
  */
-async function waitForPublishedEvents<E extends NetworkControllerEvents>(
+async function waitForPublishedEvents<E extends NetworkControllerEvents>({
+  messenger,
+  eventType,
+  count: expectedNumberOfEvents = 1,
+  filter: isEventPayloadInteresting = () => true,
+  wait: timeBeforeAssumingNoMoreEvents = 150,
+  produceEvents = () => {
```

Comment:
This reminds me, another change we may want to make (in another PR) is to rename this to `operation`.

Reply:
Good idea! I will work on that next